### DISCUSSION TITLE
NX ORIN - HDMI/DISPLAY not consistently detected

### DISCUSSION THREAD

**Discussion by andregirard37** [OE4T/meta-tegra#1625] [Q&A]
Hello,
I am having trouble fixing an issue with the DISPLAY on the NX Orin devkit board. I am using Kirkstone latest (5.1.3).
The symptoms are as follows:
When I type "who", I get 3 possible outcome:s
```
user   ttyTCU0      Mar  9 12:35
root     pts/1        Mar  9 12:34 (:1)
```

```
user   ttyTCU0      Mar  9 12:38
root     pts/1        Mar  9 12:37 (:0)
```

And sometimes, the HDMI seems to not be detected as I don't see the prompt on the monitor and I get:
`user   ttyTCU0      Mar  9 13:11`

So in that last case, no DISPLAY (:0 or :1)

For our application, we rely on OpenGL and we need to export DISPLAY=:0 , or :1, but we need it to be deterministic. We don't have the problem with the NX Xavier, also 5.1.3, but for some reason on the Orin we do have the problem, bouncing from :0 to :1 to nothing.
When I get :0, I see in /var/log/ the files Xorg.0.log and Xorg.0.log.old, which appear to be similar.
When I get :1, I see in /var/log/ the files Xorg.0.log and Xorg.1.log, which also appear to be similar.
When I get nothing, I see in /var/log/ the files Xorg.0.log and Xorg.1.log, and this time Xorg.1.log has the error message:

```
...
[    10.688] (--) using VT number 3

[    16.860] (WW) xf86OpenConsole: VT_WAITACTIVE failed: Input/output error
[    16.860] (EE) 
Fatal server error:
[    16.860] (EE) xf86OpenConsole: Switching VT failed
[    16.860] (EE) 
[    16.860] (EE) 
Please consult the The X.Org Foundation support 
         at http://wiki.x.org
 for help. 
[    16.860] (EE) Please also check the log file at "/var/log/Xorg.1.log" for additional information.
[    16.860] (EE) 
[    16.860] (EE) Server terminated with error (1). Closing log file.
```

When I have :1, I noticed that typing xrandr gives
```
root@User-NX:~# xrandr
Can't open display :0
```

And xinit gives the following:
```
root@User-NX:~# xinit

X.Org X Server 1.21.1.8
X Protocol Version 11, Revision 0
Current Operating System: Linux User-NX 5.10.192-l4t-35.5.0+gdb857ecdbe10 #1 SMP PREEMPT Wed Feb 28 11:28:56 UTC 2024 aarch64
Kernel command line: mminit_loglevel=4 console=ttyTCU0,115200 firmware_class.path=/etc/firmware fbcon=map:0 net.ifnames=0 nospectre_bhb 
 
Current version of pixman: 0.40.0
        Before reporting problems, check http://wiki.x.org
        to make sure that you have the latest version.
Markers: (--) probed, (**) from config file, (==) default setting,
        (++) from command line, (!!) notice, (II) informational,
        (WW) warning, (EE) error, (NI) not implemented, (??) unknown.
(==) Log file: "/var/log/Xorg.0.log", Time: Fri Mar  9 14:03:59 2018
(==) Using config file: "/etc/X11/xorg.conf"
(==) Using config directory: "/etc/X11/xorg.conf.d"
(==) Using system config directory "/usr/share/X11/xorg.conf.d"
The XKEYBOARD keymap compiler (xkbcomp) reports:
> Warning:          Could not resolve keysym XF86EmojiPicker
Errors from xkbcomp are not fatal to the X server
Warning: locale not supported by Xlib, locale set to C
Warning: X locale modifiers not supported, using default
xterm: Failed to open input method
```

When I have DISPLAY = :0, then xrandr and xinit seems to work fine.

Any idea what could be wrong in any of the files (dtb, etc) and any possible fix ?
Thank you!
Andre

**Comment by madisongh** [comment]
Since the Orin dev kit only has DisplayPort output, maybe this is an issue with whatever DP-HDMI adapter you're using? I seem to recall some threads in the NVIDIA dev forum about monitor detection problems.  There's also [this section](https://docs.nvidia.com/jetson/archives/r35.5.0/DeveloperGuide/SD/Kernel/DisplayConfigurationAndBringUp.html) of the L4T documentation that might help.

**Reply by andregirard37** [reply]
Thanks for the reply. We originally used the SEEED Studio J4012 devkit and now our own board. Both have the issue.
I was able to only now have either no display detected, or display :1. Not sure why never :0 but anyway... I also noticed the following extra messages in the dmesg when it does not work (no display detected):
[   15.077346] ------------[ cut here ]------------
[   15.077395] WARNING: CPU: 1 PID: 820 at drivers/tty/vt/vt_ioctl.c:244 vt_waitactive+0x0
[   15.077399] Modules linked in: uvcvideo nvidia_drm(O) nvidia_modeset(O) kbuf(O) pwm_fa)
[   15.077472] CPU: 1 PID: 820 Comm: X Tainted: G           O      5.10.192-l4t-35.5.0+gd1
[   15.077476] Hardware name: NVIDIA Orin NX Developer Kit (DT)
[   15.077483] pstate: 60400009 (nZCv daif +PAN -UAO -TCO BTYPE=--)
[   15.077489] pc : vt_waitactive+0x210/0x2a0
[   15.077494] lr : vt_waitactive+0x20c/0x2a0
[   15.077498] sp : ffff800012853be0
[   15.077501] x29: ffff800012853c50 x28: dead000000000100
[   15.077508] x27: dead000000000122 x26: 0000000000000001
[   15.077514] x25: ffffbe3e71607248 x24: 0000000000000003
[   15.077520] x23: ffffbe3e71476ea0 x22: ffffbe3e71476e90
[   15.077526] x21: ffff800012853c18 x20: ffffbe3e71606ef8
[   15.077532] x19: 0000000000000000 x18: 0000000000000000
[   15.077537] x17: 0000000000000000 x16: 0000000000000000
[   15.077544] x15: 0000000000000000 x14: ffff6efb801d1d80
[   15.077549] x13: ffffb0c07d8fa000 x12: 000000003464d91d
[   15.077555] x11: 0000000000000000 x10: 0000000000000ab0
[   15.077560] x9 : ffff800012853a90 x8 : ffff6efb88eb7250
[   15.077564] x7 : ffffbe3e70db5000 x6 : 0000000036fcd479
[   15.077568] x5 : 00000000410fd420 x4 : 0000000000c0000e
[   15.077571] x3 : ffffbe3e71476e90 x2 : ffffbe3e6f7b0d70
[   15.077575] x1 : ffff6efb88eb6740 x0 : 0000000100000000
[   15.077580] Call trace:
[   15.077585]  vt_waitactive+0x210/0x2a0
[   15.077589]  vt_ioctl+0x558/0x1334
[   15.077595]  tty_ioctl+0x27c/0xd50
[   15.077603]  __arm64_sys_ioctl+0xac/0xec
[   15.077612]  el0_svc_common.constprop.0+0x80/0x1c4
[   15.077615]  do_el0_svc+0x74/0x9c
[   15.077624]  el0_svc+0x1c/0x2c
[   15.077627]  el0_sync_handler+0x9c/0x120
[   15.077631]  el0_sync+0x16c/0x180
[   15.077634] ---[ end trace f796e3abab22da4a ]---
